### PR TITLE
feat(metrics): add /metrics which returns prometheus metrics

### PR DESCRIPTION
**Pull request by co42**
Will be useful for Dedicated Endpoints to improve request forwarding fairness and scaling

### CODE REVIEW

File: src/huggingface_inference_toolkit/webservice_starlette.py [hunk 1]
```diff
@@ -132,6 +144,7 @@ async def predict(request):
         routes=[
             Route(_health_route, health, methods=["GET"]),
             Route(_predict_route, predict, methods=["POST"]),
+            Route("/metrics", metrics, methods=["GET"]),
```
Comment (co42) [inline src/huggingface_inference_toolkit/webservice_starlette.py:147]: Do we need it in AIP_MODE ?

File: src/huggingface_inference_toolkit/webservice_starlette.py [hunk 2]
```diff
@@ -69,6 +69,18 @@ async def health(request):
...
 
+# Report Prometheus metrics
+# inf_batch_current_size: Current number of requests being processed
+# inf_queue_size: Number of requests waiting in the queue
+async def metrics(request):
+    batch_current_size = MAX_CONCURRENT_THREADS - MAX_THREADS_GUARD.value
+    queue_size = MAX_THREADS_GUARD.statistics().tasks_waiting
+    return PlainTextResponse(
+        f"inf_batch_current_size {batch_current_size}\n" +
+        f"inf_queue_size {queue_size}\n"
```
Comment (co42) [inline src/huggingface_inference_toolkit/webservice_starlette.py:80]: Not sure the naming is the best, I copied TGI

File: src/huggingface_inference_toolkit/webservice_starlette.py (hunk 1 above)
Comment (alvarobartt) [reply src/huggingface_inference_toolkit/webservice_starlette.py:147]: Hi here! So the short answer is no we don't, but here's more info on why we don't:

The `AIP_MODE` is the one suited specifically for Google Vertex AI (AIP stands for AI Platform on Google Cloud), and there just the `health` and `predict` endpoints are being actively used (set within Vertex AI via the `AIP_HEALTH_ROUTE` and `API_PREDICT_ROUTE` environment variables)

🤗

**Review by alvarobartt — APPROVED** [review]
LGTM, maybe let's wait and see if @philschmid agrees on the metrics reported and we can merge 🤗 Thanks!

File: src/huggingface_inference_toolkit/webservice_starlette.py (hunk 1 above)
Comment (co42) [reply src/huggingface_inference_toolkit/webservice_starlette.py]: Got it, I removed it 👍

File: src/huggingface_inference_toolkit/webservice_starlette.py (hunk 2 above)
Comment (philschmid) [reply src/huggingface_inference_toolkit/webservice_starlette.py:80]: Are we okay that this will always be 1?

**Review by philschmid — APPROVED** [review]
LGTM

File: src/huggingface_inference_toolkit/webservice_starlette.py (hunk 2 above)
Comment (co42) [reply src/huggingface_inference_toolkit/webservice_starlette.py:80]: If it bother you or makes it unclear for customers I can remove `inf_batch_current_size`

**Pull request merged**